standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -914,6 +914,7 @@ private boolean isViewTable(String catName, String dbName, String tblName) throw
     long queryTime = doTrace ? System.nanoTime() : 0;
     MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
     if (sqlResult.isEmpty()) {
+      query.closeAll();
       return Collections.emptyList(); // no partitions, bail early.
     }
```

Comment:
Can you just remove this entire block?  This will allow for a single return statement at the bottom and also allows the cleanup to happen.